packages/context/src/context.ts

```diff
@@ -161,27 +252,19 @@ export class Context {
    * include the binding or `false` to exclude the binding.
    */
   find<ValueType = BoundValue>(
-    filter: (binding: Readonly<Binding<ValueType>>) => boolean,
+    filter: BindingFilter,
   ): Readonly<Binding<ValueType>>[];
 
   find<ValueType = BoundValue>(
-    pattern?:
-      | string
-      | RegExp
-      | ((binding: Readonly<Binding<ValueType>>) => boolean),
+    pattern?: string | RegExp | BindingFilter,
   ): Readonly<Binding<ValueType>>[] {
-    let bindings: Readonly<Binding>[] = [];
-    let filter: (binding: Readonly<Binding>) => boolean;
-    if (!pattern) {
-      filter = binding => true;
-    } else if (typeof pattern === 'string') {
-      const regex = this.wildcardToRegExp(pattern);
-      filter = binding => regex.test(binding.key);
-    } else if (pattern instanceof RegExp) {
-      filter = binding => pattern.test(binding.key);
-    } else {
-      filter = pattern;
-    }
+    const bindings: Readonly<Binding>[] = [];
+    const filter: BindingFilter =
+      pattern == null ||
+      typeof pattern === 'string' ||
+      pattern instanceof RegExp
```

Comment:
Can we simplify this condition by checking whether `pattern` is already a BindingFilter?

```ts
const filter: BindingFilter = typeof pattern === 'function' 
  ? pattern 
  : Context.bindingKeyFilter(pattern);
```

Or maybe even better, modify `bindingKeyFilter` to accept a filter function too, so that places accepting string/RegExp/BindingFilter don't have to treat BindingFilter values differently.

Reply:
> Can we simplify this condition by checking whether `pattern` is already a BindingFilter?
>
> ```ts
> const filter: BindingFilter = typeof pattern === 'function' 
>   ? pattern 
>   : Context.bindingKeyFilter(pattern);
> ```
>
> Or maybe even better, modify `bindingKeyFilter` to accept a filter function too, so that places accepting string/RegExp/BindingFilter don't have to treat BindingFilter values differently.

PTAL ☝️